epochStart/bootstrap/startInEpochScheduled.go

```diff
@@ -439,9 +438,28 @@ func (ses *startInEpochWithScheduledDataSyncer) getScheduledTransactionHashes(he
 	}
 
 	scheduledTxs := make(map[string]uint32)
-	for _, mb := range miniBlocks {
-		for _, txHash := range mb.TxHashes {
-			scheduledTxs[string(txHash)] = mb.GetReceiverShardID()
+	for _, miniBlockHeader := range miniBlockHeaders {
+		miniBlockHash := miniBlockHeader.GetHash()
+		miniBlock, ok := miniBlocks[string(miniBlockHash)]
+		if !ok {
+			log.Warn("startInEpochWithScheduledDataSyncer.getScheduledTransactionHashes: mini block was not found", "mb hash", miniBlockHash)
+			continue
+		}
+
+		firstIndex := miniBlockHeader.GetIndexOfFirstTxProcessed()
+		lastIndex := miniBlockHeader.GetIndexOfLastTxProcessed()
```

Comment:
If this will happen the below for will be not executed at all, and the scheduled txs returned will not match and the node will be stuck to process the first block in the new epoch. The same happens if I would add this sanity check here and return in this point. I will add this check only to display a warn here.